types/use-resize-observer/index.d.ts

```diff
@@ -0,0 +1,101 @@
+// Type definitions for use-resize-observer 5.0
+// Project: https://github.com/ZeeCoder/use-resize-observer#readme
+// Definitions by: PetrSvirak <https://github.com/PetrSvirak>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.5
+export {};
+
+/***
+ * Allows using a ResizeObserver to measure size of an element assigned with ref returned from the hook.
+ * The hook is invoked at least once before the first measurement of actual size by the observer.
+ * @return a ref,
+ *         and ref.current's width (1 before the first measurement),
+ *         and ref.current's height (1 before the first measurement).
+ */
+export default function(): ObserverResultWithDefaultSize;
```

Comment:
The source library has a `main` entry pointing to a CommonJS module and a `module` entry pointing to an ESM module. You’ve correctly modeled the ESM module here. But, TypeScript doesn’t yet have the ability to model both formats side-by-side for the same module, so unfortunately we have to pick one and ignore the other. For now, I think that should be the `main` entry, so this should be written as a CJS module:

```ts
declare function useResizeObserver(): ObserverResultWithDefaultSize;
declare function useResizeObserver(// ... continue with overloads...

export = useResizeObserver;
```

Users can still do `import useResizeObserver from 'use-resize-observer'` if they have `esModuleInterop` turned on.

Reply:
Makes sense, thanks 😃. 
I did not find a way to export additional interfaces, though 😢. So, I had to remove `ObserverDefaults` and omit exports of `ObserverResult*` from `index.d.ts`. I preserved tests for `ObserverDefaults` even though I had to repeat some types from `index.d.ts` in `the use-resize-observer-tests.ts` – while it is a bit more error-prone, I found the tests quite useful while defining types/expressing possible defaults' variations. See https://github.com/DefinitelyTyped/DefinitelyTyped/commit/e4ef59a31c4cbb0c4ec7f9b3f367cdb77598d10b.

Reply:
Oh, here’s how you do that:

```ts
declare function useResizeObserver(): useResizeObserver.ObserverResultWithDefaultSize;
declare function useResizeObserver(// ... continue with overloads...

declare namespace useResizeObserver {
  export interface ObserverResultWithDefaultSize { ... }
}

export = useResizeObserver;
```

Reply:
Thanks 👍, interfaces put back in https://github.com/DefinitelyTyped/DefinitelyTyped/commit/370bfedb863aca89cacee499dfef7685cf5c775a.